Sources/Afterpay/Checkout/CheckoutV2ViewController.swift

```diff
@@ -0,0 +1,341 @@
+    self.view = view
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    if #available(iOS 13.0, *) {
+      overrideUserInterfaceStyle = .light
+    } else {
+      navigationItem.rightBarButtonItem = UIBarButtonItem(
+        title: "Cancel",
+        style: .plain,
+        target: self,
+        action: #selector(presentCancelConfirmation)
+      )
+    }
+
+    loadingWebView?.loadHTMLString(StaticContent.loadingHTML, baseURL: nil)
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+
+    let webView: WKWebView = bootstrapWebView
+    let load: () -> Void = { [bootstrapURL] in webView.load(URLRequest(url: bootstrapURL)) }
+
+    // Remove bootstrap disk caches so that the latest bootstrap is loaded
+    let dataStore = webView.configuration.websiteDataStore
+    let dataTypes: Set<String> = [WKWebsiteDataTypeDiskCache]
+    let displayName = configuration.environment.bootstrapCacheDisplayName
+
+    dataStore.fetchDataRecords(ofTypes: dataTypes) { records in
+      let bootstrapRecords = records.filter { record in record.displayName == displayName }
+      dataStore.removeData(ofTypes: dataTypes, for: bootstrapRecords, completionHandler: load)
+    }
+  }
+
+  // MARK: UIAdaptivePresentationControllerDelegate
+
+  func presentationControllerDidAttemptToDismiss(
+    _ presentationController: UIPresentationController
+  ) {
+    presentCancelConfirmation()
+  }
+
+  // MARK: Actions
+
+  @objc private func presentCancelConfirmation() {
+    let actionSheet = Alerts.areYouSureYouWantToCancel {
+      self.dismiss(animated: true) { self.completion(.cancelled(reason: .userInitiated)) }
+    }
+
+    present(actionSheet, animated: true, completion: nil)
+  }
+
+  // MARK: WKNavigationDelegate
+
+  func webView(
+    _ webView: WKWebView,
+    decidePolicyFor navigationAction: WKNavigationAction,
+    decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
+  ) {
+    guard let url = navigationAction.request.url, navigationAction.targetFrame == nil else {
+      return decisionHandler(.allow)
+    }
+
+    decisionHandler(.cancel)
+    UIApplication.shared.open(url)
+  }
+
+  func webView(
+    _ webView: WKWebView,
+    didFailProvisionalNavigation navigation: WKNavigation!,
+    withError error: Error
+  ) {
+    handleError(webView: webView, error: error)
+  }
+
+  func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+    if webView == checkoutWebView {
+      checkoutWebView?.isHidden = false
+      loadingWebView?.removeFromSuperview()
+      loadingWebView = nil
+    } else if webView == bootstrapWebView {
+      commenceCheckout()
+    }
+  }
+
+  private func commenceCheckout() {
+    guard let didCommenceCheckout = didCommenceCheckout else {
+      return assertionFailure(
+        "For checkout to function you must set `didCommenceCheckout` via either "
+          + "`Afterpay.presentCheckoutV2Modally` or `Afterpay.setCheckoutV2Handler`"
```

Comment:
I wonder why checkout handler can be set in two different places?

Reply:
You can either handle it inline or in a separate object

Reply:
I mean, why do we provide alternative way to set the handler (separate object)? 
Is this to shorten the parameters?

Reply:
It can provide a shorter callsite. Also Android can only be done through a separate object so it provides a way to keep both code-bases somewhat in sync